lightning/src/ln/reorg_tests.rs

```diff
@@ -180,3 +188,185 @@ fn test_onchain_htlc_claim_reorg_remote_commitment() {
 fn test_onchain_htlc_timeout_delay_remote_commitment() {
 	do_test_onchain_htlc_reorg(false, false);
 }
+
+fn do_test_unconf_chan(reload_node: bool) {
```

Comment:
Don't think it makes _too_ much of a difference, but we could add a test that's closer to a real-world scenario if we tested deserializing and *then* syncing all the new blocks to tip (thus generating a `BackgroundEvent`). (And then `watch_channel`'d and `process_background_events`'d, etc.)

Reply:
I kinda cheated and expanded the test to either do the disconnect before or after the reload.